theano/tensor/basic.py

```diff
@@ -3471,6 +3479,12 @@ def _make_node_internal(self, axis, tensors,
                 bcastable = [False] * len(
                     as_tensor_variable_args[0].type.broadcastable)
 
+                axis = theano.tensor.switch(
```

Comment:
Up to now, we always do those check in the op, not in the graph. Even if we duplicate this check 4 times, I think it is better, it is not a natural place to put it in the graph.